optics/src/Optics.hs

```diff
@@ -858,85 +858,112 @@ import Data.Either.Optics                    as P
--- +--------------------+---------------------------------------------------+
--- | 'itraversed'       | Build an 'IxTraversal' from the                   |
--- |                    | 'TraversableWithIndex' class.                     |
--- +--------------------+---------------------------------------------------+
--- | 'itraverseOf'      | Update target(s) with an 'Applicative', with      |
--- |                    | access to the index.                              |
--- +--------------------+---------------------------------------------------+
-
+-- +-----------------------------------+---------------------------------------------------+
+-- |Combinator                         |Notes                                              |
+-- +===================================+===================================================+
+-- |                                                                                       |
+-- +-----------------------------------+---------------------------------------------------+
+-- |__Getters__                                                                            |
+-- +-----------------------------------+---------------------------------------------------+
+-- |'to'                               |Build a 'Getter' from a plain function.            |
+-- +-----------------------------------+---------------------------------------------------+
+-- |'view' / '^.'                      |View a single target.                              |
+-- +-----------------------------------+---------------------------------------------------+
+-- |'views'                            |View after applying a function.                    |
+-- +-----------------------------------+---------------------------------------------------+
+-- |                                                                                       |
+-- +-----------------------------------+---------------------------------------------------+
+-- |__Setters__                                                                            |
+-- +-----------------------------------+---------------------------------------------------+
+-- |'sets'                             |Build a 'Setter' from an update function.          |
+-- +-----------------------------------+---------------------------------------------------+
+-- |'mapped'                           |Build a 'Setter' from the 'Functor' class.         |
+-- +-----------------------------------+---------------------------------------------------+
+-- |'set' / '.~'                       |Replace target(s) with value.                      |
+-- +-----------------------------------+---------------------------------------------------+
+-- |'over' / '%~'                      |Replace target(s) by applying a function.          |
+-- +-----------------------------------+---------------------------------------------------+
+-- |'assign' /                         |'Control.Monad.State.MonadState' version of 'set'; |
```

Comment:
i thought of making third column as in `lens` table; that would highlight the relationship between `%~` and `%=` etc. Yet, I don't know how to combine notes reasonably

Reply:
Yes, I wondered about that but wasn't very keen on adding extra columns just for two rows to have stateful variants. Perhaps they should appear in a completely different table?

Reply:
@adamgundry otoh, I actually didn't realise we don't have `use`, I didn't need it yet myself, but surely some `MonadState` / `Zoom` will ask for it. 

I.e. I'm not sure how long it will stay at only two rows :)

Reply:
I've made a new table instead, which I think looks better as it illustrates the relationships more clearly.

Reply:
Indeed :+1:

Reply:
> I actually didn't realise we don't have use

We do, it's in `Optics.View`.

Reply:
> We do, it's in Optics.View.

ou. "this is terrible". It should be `guse` if it's there, and have proper `use`, `preuse` (someething for list), without `lens`-weirdness.

Reply:
PRs welcome ;)

Reply:
Let's move this to #261.